Marlin/src/gcode/lcd/M0_M1.cpp

```diff
@@ -80,7 +80,14 @@ void GcodeSuite::M0_M1() {
 
   #elif ENABLED(EXTENSIBLE_UI)
 
-    ExtUI::onUserConfirmRequired(has_message ? args : MSG_USERWAIT); // SRAM string
+    if (has_message)
+      ExtUI::onUserConfirmRequired(args); // SRAM string
+    else {
+      PGM_P pstr = GET_TEXT(MSG_USERWAIT);
+      char str[strlen_P(pstr) + 1];
+      strcpy_P(str, pstr);
+      ExtUI::onUserConfirmRequired(str);
+    }
```

Comment:
Apparently `onUserConfirmRequired` expects a `PROGMEM` string. (All other instances of it  Should we rename it `onUserConfirmRequired_P` and then add a `onUserConfirmRequired` method taking SRAM strings?

Reply:
In a lot of my UI code, I use "progmem_str" for PROGMEM strings, and "const char *" for SRAM strings. Not only does this make it clearer which is expected, but it also allows for overloading which is important for template classes or methods. You cannot make generic methods when using "_P" prefixes.

Anyhow, `ExtUI::onUserConfirmRequired()` cannot expect a PROGMEM string, because the message is an arbitrary string set from a GCODE command and isn't known at startup. So I'm not sure where you got the idea it is using a PROGMEM string. At least the LULZBOT_UI is properly implementing it as a SRAM string. If UltraLCD is implementing it as PROGMEM, then this would indeed be a problem.

Reply:
Within Marlin we've been using `PGM_P` for `PROGMEM` strings. Same as `progmem_str` essentially.

Reply:
> ExtUI::onUserConfirmRequired() cannot expect a PROGMEM string, because the message is an arbitrary string set from a GCODE command

If you search for it in Marlin you will find that only one piece of code uses it to pass an SRAM string. Every other instance passes a `PSTR`.

Reply:
I see that your `progmem_str` is using `__FlashStringHelper`.  I would like to explore the difference between the `__FlashStringHelper` and dumb old `PSTR` in terms of program memory and ram cost. We have been assiduously avoiding the use of `__FlashStringHelper` up to now, and I would like to know if our reticence is justified.